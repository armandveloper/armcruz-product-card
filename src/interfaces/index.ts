import { Props as ProductCardProps } from '../components/ProductCard';
import { Props as ProductTitleProps } from '../components/ProductTitle';
import { Props as ProductImageProps } from '../components/ProductImage';
import { Props as ProductButtonsProps } from '../components/ProductButtons';

export interface ProductContextProps extends ProductCounter {
	product: Product;
	isMaxCountReached: boolean;
}

export interface Product {
	id: string;
	img?: string;
	title: string;
}

export interface ProductCounter {
	counter: number;
	isMaxCountReached: boolean;
	maxCount?: number;
	increaseBy: (value: number) => void;
	reset: () => void;
}

export interface ProductCardHOCProps {
	(props: ProductCardProps): JSX.Element;
	Title: (props: ProductTitleProps) => JSX.Element;
	Image: (props: ProductImageProps) => JSX.Element;
	Buttons: (props: ProductButtonsProps) => JSX.Element;
}

export interface OnChangeArgs {
	product: Product;
	count: number;
}

export interface CartState {
	[key: string]: ProductInCart;
}

export interface ProductInCart extends Product {
	count: number;
}

export interface IInitialValues {
	count?: number;
	maxCount?: number;
}

export interface ProductCardHandlers {
	count: number;
	isMaxCountReached: boolean;
	maxCount?: number;
	product: Product;

	increaseBy: (value: number) => void;
	reset: () => void;
}
