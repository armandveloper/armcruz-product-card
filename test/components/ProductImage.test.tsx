import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../src';
import { product2 } from '../data';

describe('ProductImage', () => {
  test('can render the component with the prop img', () => {
    const wrapper = renderer.create(<ProductImage img="/coffee-mug2.png" />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('can render the component with the img from provider', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>
        {() => <ProductCard.Image />}
      </ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
