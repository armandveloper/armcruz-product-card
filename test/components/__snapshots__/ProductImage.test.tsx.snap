// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage can render the component with the img from provider 1`] = `
<div
  className="productCard "
>
  <img
    alt=""
    className="productImg "
    src="/coffee-mug.png"
  />
</div>
`;

exports[`ProductImage can render the component with the prop img 1`] = `
<img
  alt=""
  className="productImg "
  src="/coffee-mug2.png"
/>
`;
